app/src/behaviors/behavior_macro.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_macro
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <zmk/behavior.h>
+#include <zmk/behavior_queue.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+enum behavior_macro_mode {
+    MACRO_MODE_TAP,
+    MACRO_MODE_PRESS,
+    MACRO_MODE_RELEASE,
+};
+
+struct behavior_macro_trigger_state {
+    uint32_t wait_ms;
+    uint32_t tap_ms;
+    enum behavior_macro_mode mode;
+    int8_t start_index;
+    int8_t end_index;
+};
+
+struct behavior_macro_state {
+    struct behavior_macro_trigger_state release_state;
+
+    int8_t last_press_binding;
+};
+
+struct behavior_macro_config {
+    uint32_t default_wait_ms;
+    uint32_t default_tap_ms;
+    uint8_t count;
+    struct zmk_behavior_binding bindings[];
+};
+
+#define TAP_MODE DT_LABEL(DT_INST(0, zmk_macro_control_mode_tap))
+#define PRESS_MODE DT_LABEL(DT_INST(0, zmk_macro_control_mode_press))
+#define REL_MODE DT_LABEL(DT_INST(0, zmk_macro_control_mode_release))
+
+#define TAP_TIME DT_LABEL(DT_INST(0, zmk_macro_control_tap_time))
+#define WAIT_TIME DT_LABEL(DT_INST(0, zmk_macro_control_wait_time))
+#define WAIT_REL DT_LABEL(DT_INST(0, zmk_macro_pause_for_release))
+
+#define _IS_NODE_MATCH(a, b) (strcmp(a, b) == 0)
```

Comment:
Any symbol starting with an underscore followed by a capital letter is technically reserved for the compiler in C.

Reply:
Thanks. Renamed.